dynawo/sources/Solvers/AlgebraicSolvers/DYNSolverKINAlgRestoration.cpp

```diff
@@ -292,7 +290,11 @@ SolverKINAlgRestoration::solve(bool noInitSetup) {
 
   // first evaluation of F in order to fill the scaling vector
   firstIteration_ = true;
-  model_->evalF(t0_, &y0_[0], &yp0_[0], &F_[0]);
+  if (evaluateOnlyModeAtFirstIter)
+    model_->evalFMode(t0_, &y0_[0], &yp0_[0], &F_[0]);
+  else
+    model_->evalF(t0_, &y0_[0], &yp0_[0], &F_[0]);
+  model_->reinitMode();
```

Comment:
I don't think you need to reinit the mode here because you haven't modified it into the process.

Reply:
it is needed as it is not done anymore in the solvers. I could put it back there if you prefer.

Reply:
I missed it during the first look.
But yes, I would prefer having it there after the solve.

Reply:
OK

Reply:
Solved